chapter06/src/main/java/com/seaofnodes/simple/Parser.java

```diff
@@ -200,7 +200,7 @@ private Node parseIf() {
     private Node parseReturn() {
         var expr = require(parseExpression(), ";");
         Node ret = STOP.addReturn(new ReturnNode(ctrl(), expr).peephole());
-        ctrl(null);             // Kill control
```

Comment:
Like this change, setting ctrl to null was causing issues such as null input to Region